mininet/link.py

```diff
@@ -100,8 +115,18 @@ def updateIP( self ):
         ifconfig, _err, _exitCode = self.node.pexec(
             'ifconfig %s' % self.name )
         ips = self._ipMatchRegex.findall( ifconfig )
+        ip6s = self._ip6MatchRegex.findall( ifconfig )
         self.ip = ips[ 0 ] if ips else None
-        return self.ip
+        self.ip6 = None
+        self.ip6ll = None
+        for addr, scope in ip6s:
+            if scope == 'link':
+                if self.ip6ll is None:
+                    self.ip6ll = addr
+            else:
+                if self.ip6 is None:
+                    self.ip6 = addr
```

Comment:
In addition to identifying the IPv4 address (if any), also identify a global and link-local IPv6 address.

Note that the updated method is mostly backwards compatible with the previous version, with the exception of the return value.  However, that is _mostly_ backwards compatible as well.  See changes to `Link.IP()`